sqlglot/optimizer/optimizer.py

```diff
@@ -10,8 +11,23 @@
 from sqlglot.optimizer.quote_identities import quote_identities
 from sqlglot.optimizer.unnest_subqueries import unnest_subqueries
 
+RULES = (
+    qualify_tables,
+    isolate_table_selects,
+    qualify_columns,
+    pushdown_projections,
+    normalize,
+    unnest_subqueries,
+    expand_multi_table_selects,
+    pushdown_predicates,
+    optimize_joins,
+    eliminate_subqueries,
+    merge_derived_tables,
```

Comment:
perhaps merging should happen before isolate table selects, so that merging remove reduntant

select * from (select *), but then isolate still works

Reply:
merge_derived_tables depends on qualify_columns.

Also, in my use case, we don't want isolated table selects. I actually only left it in here so `eliminate_subqueries` can do its thing.

side note: should we be making the rule dependencies explicit? We might be able to automatically order them, which would make it easier to pick and choose rules.

Reply:
maybe later on the rules dependencies, but no need to overcomplicate for now.

i think this comes down to what the default should be